src/CVRF/cvrf_priv.c

```diff
@@ -0,0 +1,1518 @@
+				remed = cvrf_remediation_parse(reader);
+				if (remed != NULL)
+					cvrf_vulnerability_add_remediation(vulnerability, remed);
+				xmlTextReaderNextNode(reader);
+			}
+		}
+
+		xmlTextReaderNextNode(reader);
+	}
+
+	return vulnerability;
+}
+
+struct cvrf_remediation *cvrf_remediation_parse(xmlTextReaderPtr reader) {
+	__attribute__nonnull__(reader);
+
+	struct cvrf_remediation *remed = NULL;
+	remed = cvrf_remediation_new();
+	if (remed == NULL)
+		return NULL;
+
+	remed->remed_type = cvrf_remediation_type_parse(reader, "Type");
+	remed->remed_date = (char *)xmlTextReaderGetAttribute(reader, BAD_CAST "Date");
+	xmlTextReaderNextElement(reader);
+
+	while (xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_REMEDIATION) != 0) {
+
+		if (xmlTextReaderNodeType(reader) != XML_READER_TYPE_ELEMENT) {
+			xmlTextReaderNextNode(reader);
+			continue;
+		}
+
+		if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_REMEDIATION_DESC)) {
+			remed->remed_description = (char *)oscap_element_string_copy(reader);
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_REMEDIATION_URL)) {
+			remed->remed_URL = (char *)oscap_element_string_copy(reader);
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_PRODUCT_ID)) {
+			remed->remed_product_id = (char *)oscap_element_string_copy(reader);
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_GROUP_ID)) {
+			remed->remed_group_id = (char *)oscap_element_string_copy(reader);
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), BAD_CAST "Entitlement")) {
+			remed->remed_entitlement = (char *)oscap_element_string_copy(reader);
+		}
+
+		xmlTextReaderNextNode(reader);
+	}
+	xmlTextReaderNextNode(reader);
+
+	return remed;
+}
+
+
+struct cvrf_product_status *cvrf_product_status_parse(xmlTextReaderPtr reader) {
+
+	__attribute__nonnull__(reader);
+
+	struct cvrf_product_status *stat;
+
+	stat = cvrf_product_status_new();
+	if (stat == NULL)
+		return NULL;
+
+	stat->status_type = cvrf_product_status_type_parse(reader, "Type");
+	xmlTextReaderNextElement(reader);
+
+	while (xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_CVRF_PRODUCT_STATUS) != 0) {
+
+		if (xmlTextReaderNodeType(reader) != XML_READER_TYPE_ELEMENT) {
+			xmlTextReaderNextNode(reader);
+			continue;
+		}
+
+		if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_PRODUCT_ID)) {
+			const char *product_id = oscap_element_string_copy(reader);
+
+			if (product_id != NULL)
+				oscap_stringlist_add_string(stat->product_ids, product_id);
+
+		}
+		xmlTextReaderNextNode(reader);
+	}
+
+	return stat;
+}
+
+
+/***************************************************************************
+ * Export functions
+ *
+ */
+void cvrf_index_export_xml(struct cvrf_index *index, struct oscap_source *export_source) {
+
+	__attribute__nonnull__(index);
+	__attribute__nonnull__(export_source);
+
+	const char *filepath = oscap_source_get_filepath(export_source);
```

Comment:
Whenever you are using `oscap_source_get_filepath` make sure that you absolutely need to. It limits the implementation and will only work with local files. I am pretty sure that in this case you don't actually need to get it as a file... We also generally use DOM for exporting because it's much more practical for putting XMLs inside XMLs.